main.go

```diff
@@ -206,6 +211,13 @@ func run(args []string, stdin io.Reader, stderr io.Writer) error {
 		return errors.New("please provide at least one pattern")
 	}
 
+	var logger *log.Logger
+	if opts.Verbose {
+		logger = log.New(os.Stdout, "", 0)
+	} else {
+		logger = log.New(io.Discard, "", 0)
+	}
```

Comment:
minor optional nit: I would have if-else switched on the `io.Writer` rather than the logger since the rest of the logger setup is the same.

```suggestion
	logOut := io.Discard
	if opts.Verbose {
		logOut = os.Stdout
	}
	log := log.New(logOut, "", 0)
```

Also, when using the stdlib log package, it's generally preferable to shadow the package (`log := log.New(..)`). This makes sure you don't accidentally use the top-level `log.Printf` functions (of the default logger) rather than methods on your new `*log.Logger`.

---

Separately, do we want verbose output on stdout or stderr? Generally, if the output is meant to be machine processable, it should be stdout so that it's easy to pipe into other commands, but if it's largely for human consumption, it should be stderr. (See also [this part of clig.dev](https://clig.dev/#the-basics).) In this case, I'm open to either.